File: 141. Linked List Cycle/LinkedListCycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 * @author:leacoder
 * @des:  存储记录 环形链表 HashSet 
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        HashSet<ListNode> hSet = new HashSet<>();
        while(null != head){
            if(hSet.contains(head)){
                return true;
            }else{
                hSet.add(head);
            }
            head = head.next;
        }
        return false;
    }
}

File: 141. Linked List Cycle/LinkedListCycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 * @author:leacoder
 * @des: 快慢指针 环形链表 
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode *slow=head,*fast=head;
        while(NULL!=slow&&NULL!=fast&&NULL!=fast->next){
            slow = slow->next;
            fast = fast->next->next;
            if(slow == fast) 
                return true;
        }
        return false;
    }
};